db/schema.rb

```diff
@@ -322,6 +323,7 @@
     t.boolean "gametime_gold", default: false, null: false
     t.boolean "gametime_reduced", default: false, null: false
     t.boolean "gametime_skipped", default: false, null: false
+    t.string "icon"
```

Comment:
Currently just saving the icon to the DB.  Is it preferable to use ActiveStorage for this instead?  They're encoded as base something strings in LiveSplit but I'm sure the conversion is easy enough.

@CryZe I remember you saying something about shrinking the images, is that something we should be doing before we save them?

Reply:
The latest released livesplit-core automatically shrinks the images when parsing the splits, so when you get them from the Run, they are already shrinked. However the base64 storage is not the case anymore, you get the raw image bytes now (I think it still is in 0.11, which is the latest version. The change is only on the master branch atm)

Reply:
Hm, ok.  Will let @glacials decide one what he wants to do with the images.

Thanks for the heads up appreciate it.

Reply:
I strongly prefer storing in Active Storage; I'm permanently scarred from storing non-text and/or variable-length stuff in the database after doing it early on in Splits.io. Even if livesplit-core guarantees some max size, it'll still be a long-term pain to fight against Rails's opinions on what a database field should be.

Active Storage also has some nice [convenience methods for displaying and transforming images][1]. Even if they're already shrunken to "not ridiculous size", we can use these to shrink them even more depending on how they're being displayed. It will also make it easier for us to let users modify images if we ever decide we want to.

But really, most of the benefit is from not needing to twist Rails's arm to behave sanely.

[1]: https://guides.rubyonrails.org/active_storage_overview.html#transforming-images

Reply:
This was the right call.  The new variant methods allowed very easy transformations of the icons.